pkgs/tools/admin/tigervnc/default.nix

```
@@ -31,14 +44,24 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  postPatch = ''
+  postPatch = lib.optionalString stdenv.isLinux ''
     sed -i -e '/^\$cmd \.= " -pn";/a$cmd .= " -xkbdir ${xkeyboard_config}/etc/X11/xkb";' unix/vncserver/vncserver.in
     fontPath=
     substituteInPlace vncviewer/vncviewer.cxx \
        --replace '"/usr/bin/ssh' '"${openssh}/bin/ssh'
 
     cp unix/xserver21.1.1.patch unix/xserver211.patch
     source_top="$(pwd)"
+  '' + ''
+    # On Mac, do not build a .dmg
+    # (remove the commands starting with hdiutil up through the second
+    #  occurrence of $VERSION.dmg)
+    # Instead, copy the .app to the source dir
+    tmpfile=$(mktemp)
+    cp release/makemacapp.in "$tmpfile"
+    awk 'BEGIN { del=0 } /hdiutil/ { del=2 } del<=0 { print } /$VERSION.dmg/ { del -= 1 }' "$tmpfile" >release/makemacapp.in
+    rm "$tmpfile"
```

```suggestion
    awk 'BEGIN { del=0 } /hdiutil/ { del=2 } del<=0 { print } /$VERSION.dmg/ { del -= 1 }' release/makemacapp.in | sponge release/makemacapp.in
```
and add more-utils/sponge to nativeBuildInputs.

---

I forgot that gawk allows for inplace modification, switched to that with latest force push rather than adding sponge if that's okay.